src/Smidge/RequestHelper.cs

```diff
@@ -93,46 +93,37 @@ public string Content(string path)
         public CompressionType GetClientCompression(IDictionary<string, StringValues> headers)
         {
             var type = CompressionType.None;
-            var agentHeader = (string)headers[HeaderNames.UserAgent];
-            if (agentHeader != null && agentHeader.Contains("MSIE 6"))
-            {
-                return type;
-            }
 
-            string acceptEncoding = headers[HeaderNames.AcceptEncoding];
-
-            if (!string.IsNullOrEmpty(acceptEncoding))
+            if (headers.TryGetValue(HeaderNames.AcceptEncoding, out StringValues acceptEncoding))
             {
-                string[] acceptedEncodings = acceptEncoding.Split(',');
-
                 // Prefer in order: Brotli, GZip, Deflate.
                 // https://www.iana.org/assignments/http-parameters/http-parameters.xml#http-content-coding-registry
-                for (var i = 0; i < acceptedEncodings.Length; i++)
+                for (var i = 0; i < acceptEncoding.Count; i++)
                 {
-                    var encoding = acceptedEncodings[i].Trim();
+                    var encoding = acceptEncoding[i].Trim();
 
                     CompressionType parsed = CompressionType.Parse(encoding);
 
                     // Brotli is typically last in the accept encoding header.
-                    if (parsed == CompressionType.Brotli)
+                    if (parsed.Equals(CompressionType.Brotli))
```

Comment:
Do we want to do any invariant checking for Equals here or no?

Reply:
No, this is a struct. I changed it to Equals instead of == because VS was warning about a not implemented == operator (seems like a VS bug as CompressionType implements the == operator as well).

Reply:
@Shazwazza Reverted these changes, if you could merge and release a new version on NuGet that'd be great 👍